### ISSUE TITLE
work flow question.

### ISSUE THREAD

**Issue by agn-7** [koide3/hdl_people_tracking#2]
Is flow works of this package (**hdl_people_tracking**)  dependent to the **hdl_localization**?   
In other words, can **hdl_people_tracking** run without depending on **hdl_localization** and detect pedestrians?

**Comment by koide3** [comment]
The standard workflow of this package requires localization. But, in case the sensor is fixed, you can run it by feeding dummy static transforms to the nodelet instead without running the localization.

**Comment by agn-7** [comment]
Thanks for the reply.
Do you have a compatible bagfile with your package that has a static Velodyne?

**Comment by koide3** [comment]
Here, a test bag for people tracking with a static velodyne.
https://drive.google.com/open?id=1oPj6xJ0VpvGc_u31EQuGAqiVMnbA5Kar

Try to launch "hdl_people_tracking_static.launch" with this bag. The first frame is treated as the background cloud.

```bash
rosparam set use_sim_time true
roslaunch hdl_people_tracking hdl_people_tracking_static.launch
```

```bash
rosbag play --clock hdl_500.bag
```

**Comment by agn-7** [comment]
In the readme file in workflow image in Related work section I see the following procedures:

[workflow](https://github.com/koide3/hdl_people_tracking/blob/master/data/figs/packages.png)

hdl_graph_slam  --> hdl_localization --> hdl_people_tracking

---
But I could not find hdl_graph_slam name in package.xml ro CMakeList.txt files in hdl_people_tracking and hdl_localization packages.

So what is the hdl_graph_slam role in this package?

**Comment by koide3** [comment]
This package requires a map cloud ("hdl_localization/data/map.pcd" in the example) to perform people tracking with a moving sensor, and in our workflow, we use hdl_graph_slam to generate the map cloud. We didn't add it to package.xml so that you can use any other SLAM algorithms instead of it.

**Comment by agn-7** [comment]
So is mean that this tracking pedestrian project works only with `hdl_400.bag` with its saved `"hdl_localization/data/map.pcd"` map? and if we want to test it on other environments we need to using the `hdl_graph_slam` or any other SLAM algorithm?

**Comment by koide3** [comment]
Yes, you need to create an environmental map with any SLAM algorithm before running this people tracking. In case you want to run it in an unmapped environment, you need to replace the background subtraction code (The package first perfoms background subtraction, and then do clustering to detect human candidates). I saw a guy modified the code so that it filters out ceiling and floor points with a pass through filter, and I think it would be helpful for you.
https://github.com/tzven0/hdl_people_tracking/commit/874cd5e66f8bb612d191185f5730e1ed7c256832

**Comment by DeepDuke** [comment]
The link for hdl_500.bag failed. Can u give me a new link for it?

**Comment by makanov001** [comment]
hi,did you got it? i meet the same problem.
